js/state/contentSettings.js

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AppDispatcher = require('../dispatcher/appDispatcher')
+const AppStore = require('../stores/appStore')
+const AppConstants = require('../constants/appConstants')
+const appConfig = require('../constants/appConfig')
+const urlParse = require('url').parse
+const siteSettings = require('./siteSettings')
+const { registerUserPrefs } = require('./userPrefs')
+
+// backward compatibility with appState siteSettings
+const parseSiteSettingsPattern = (pattern) => {
+  let normalizedPattern = pattern.replace('https?', 'https')
+  let parsed = urlParse(normalizedPattern)
+  return '[*.]' + parsed.host
+}
+
+const addContentSettings = (settingList, hostPattern, secondaryPattern = undefined, setting = 'block') => {
+  let contentSettingsPattern = parseSiteSettingsPattern(hostPattern)
+  settingList.push({
+    setting,
+    secondaryPattern,
+    primaryPattern: contentSettingsPattern
+  })
+}
+
+const getBlock3rdPartyStorage = () => {
+  return [
+    {
+      setting: 'block',
+      primaryPattern: '*',
+      secondaryPattern: '*'
+    },
+    {
+      setting: 'allow',
+      primaryPattern: '*',
+      secondaryPattern: '[firstParty]'
+    }
+  ]
+}
+
+const getAllowAllCookies = () => {
+  return [
+    {
+      setting: 'allow',
+      primaryPattern: '*',
+      secondaryPattern: '*'
+    }
+  ]
+}
+
+const getContentSettingsFromSiteSettings = (appState) => {
+  let braveryDefaults = siteSettings.braveryDefaults(appState, appConfig)
+
+  let contentSettings = {
+    cookies: braveryDefaults.cookieControl === 'block3rdPartyCookie' ? getBlock3rdPartyStorage() : getAllowAllCookies(),
+    javascript: [],
+    canvasFingerprinting: [{
```

Comment:
yea, that makes sense and would apply to ads as well. contentSettings translate the siteSettings into rules so I can use the activeSettings (which will eventually be replaced by contentSettings), but I can add an override for now